.github/workflows/shared-build-deploy-container.yml

```diff
@@ -88,10 +89,12 @@ jobs:
           echo "Using runner: $RUNNER_NAME"
           echo "RUNNER_NAME=$RUNNER_NAME" >> $GITHUB_ENV
           echo "runnerName=$RUNNER_NAME" >> $GITHUB_OUTPUT
-
+  
+  # TODO: We use the same name for the workflow step and the step inside the workflow step which creates confusion
   pick-environment:
     name: "Pick environment"
     runs-on: ubuntu-latest
+    # TODO : pick-environment-step is not used anywhere
```

Comment:
It's used in line 127:

```yml
needs.pick-environment.outputs.environment
```

Reply:
The way to setup an `output` from a job requires the step to have an id:
<https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs>

It's verbose and convoluted, fully agree with you.
But I don't think we can avoid giving the inner step an `id`..

Reply:
This was refactored. Need to be re-reviewed 😅 
Also we must use `needs`. We also renamed a bit this so maybe this isn't an issue anymore.